pkg/virt-api/rest/cpu.go

```diff
@@ -0,0 +1,46 @@
+package rest
+
+import (
+	"context"
+	"fmt"
+	"net/http"
+	"strconv"
+
+	"github.com/emicklei/go-restful/v3"
+	"k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/types"
+
+	"kubevirt.io/kubevirt/pkg/virt-api/definitions"
+)
+
+func (app *SubresourceAPIApp) CPURequestHandler(request *restful.Request, response *restful.Response) {
+	namespace := request.PathParameter(definitions.NamespaceParamName)
+	name := request.PathParameter(definitions.NameParamName)
+	cpuCountStr := request.QueryParameter(definitions.CpuCountParamName)
+
+	if cpuCountStr == "" {
+		writeError(errors.NewBadRequest("cpuCount parameter is required"), response)
+		return
+	}
+
+	cpuCount, err := strconv.Atoi(cpuCountStr)
+	if err != nil {
+		writeError(errors.NewBadRequest("cpuCount parameter must be an integer"), response)
+		return
+	}
+
+	if cpuCount < 0 {
+		writeError(errors.NewBadRequest("cpuCount parameter must be non-negative"), response)
+		return
+	}
+
+	patch := fmt.Sprintf(`[{"op": "replace", "path": "/spec/template/spec/domain/cpu/cores", "value": %d}]`, cpuCount)
```

Comment:
Please, use the [patch](https://github.com/kubevirt/kubevirt/blob/main/pkg/apimachinery/patch/patch.go) packages instead of the hardcoded operations